### PR TITLE
Use newEntity() to create DatabaseSession entities

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

**Comment by markstory** [comment]
Something is up with one of the delete tests.

**Comment by othercorey** [comment]
Yea, I don't know what it is. I had this error locally before committing.

**Comment by othercorey** [comment]
You can't reference an existing entity using `Table::newEntity()` because it doesn't pass the options to `Entity` constructor.

File: src/Http/Session/DatabaseSession.php [hunk 1]
```diff
@@ -150,14 +149,16 @@ public function write($id, $data): bool
...
+
         /** @var string $pkField */
         $pkField = $this->_table->getPrimaryKey();
-        $record[$pkField] = $id;
-        $result = $this->_table->save(new Entity($record));
+        $session = $this->_table->newEntity([
+            $pkField => $id,
+            'data' => $data,
+            'expires' => time() + $this->_timeout,
+        ], ['guard' => false]);
```
Comment (othercorey) [inline src/Http/Session/DatabaseSession.php:159]: You can't use `guard` option with `newEntity()` either since it's an Entity constructor option.

**Pull request closed without merging**